### PR TITLE
Fix build and deploy workflow

### LINKED ISSUE #163: Docs build failing (due to PyOpenGL?)
## 🐛 Bug

I just noticed that the docs build is failing on CI (https://github.com/napari/docs/actions/runs/5078127401/jobs/9122262373) when trying to render example scripts with napari. I saw a similar issue on napari-threedee and pinning `PyOpenGL < 3.1.7` fiixed the issue (https://github.com/napari-threedee/napari-threedee/pull/133). I haven't had a chance to dig in further. I'm not sure if this is something we pin here or if we need to do it in the core napari repo.

### PR DESCRIPTION
**Pull request by Czaki**
# Description

add constrains usage to build and deploy workflow 

closes #163

## Type of change

- [ ] Fixes or improves existing content
- [ ] Adds new content page(s)
- [ ] Fixes or improves workflow, documentation build or deployment

# References

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added [alt text](https://webaim.org/techniques/alttext/) to new images included in this PR

### CODE REVIEW

**Review by kevinyamauchi — APPROVED** [review]
This looks good to me. Thanks for the quick fix, @Czaki ! Somebody who has been spending more time in the docs repo might want to take a look before merging.

**Review by melissawm — APPROVED** [review]
Looks good to me, thanks @Czaki !

**Comment by kevinyamauchi** [comment]
We have two approvals, so I will merge now. Thanks for the review @melissawm !

**Pull request merged**
